File: Decorator/Java/src/condiments/SteamedMilk.java
package eu.svetlin.design.patterns.decorator.condiments;

import eu.svetlin.design.patterns.decorator.beverages.Beverage;

public class SteamedMilk extends CondimentDecorator {

    public SteamedMilk(Beverage beverage){
        this.beverage = beverage;
    }
    @Override
    public double cost() {
        return beverage.cost() + .10;
    }

    @Override
    public String getDescription() {
        return beverage.getDescription() + ", Steamed Milk";
    }
}

File: Decorator/TypeScript/src/condiments/SteamedMilk.ts
import Beverage from '../beverages/Beverage';
import CondimentDecorator from './CondimentDecorator';

class SteamedMilk extends CondimentDecorator {
  constructor(beverage: Beverage) {
    super();
    this.beverage = beverage;
  }
  getDescription() {
    return this.beverage.getDescription() + ', Steamed Milk';
  }

  cost(): number {
    return this.beverage.cost() + 0.1;
  }
}

export default SteamedMilk;
